daffodil-schematron/src/main/resources/iso-schematron-xslt2/ExtractSchFromRNG-2.xsl

```diff
@@ -0,0 +1,128 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one or more
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership.
+    The ASF licenses this file to You under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with
+    the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<!--
+	Stylesheet for extracting Schematron information from a RELAX-NG schema.
+	Based on the stylesheet for extracting Schematron information from W3C XML Schema.
+	
+	For usage details, see http://www.topologi.com/resources/schtrn_xsd_paper.html
+	
+	Created by Eddie Robertsson 2002/06/01
+	Update for ISO Schematron using XSLT2 Rick Jelliffe 2010-04-14
+	MIT licensed 2010-07-10
+-->
+
```

Comment:
Looks like this and other xsl files are licensed as both MIT and Apache? I'm not sure if that's correct. Did you add the Apache license or did you get it from Apache Camel like that?

If this is actually MIT licensed and the Apache header was incorrectly added, because we're including this in source, we must also update the root LICENSE file to include the MIT license and a listing of these files that are MIT licensed (see the "subcomponent" stuff near the end. We must also add a LICENSE file, similar to the root one but with only the stuff specific to daffodil-schematron to  ``daffodil-schematron/src/main/resrouces/METAINF/{LICENSE,NOTICE}. This license/notice is included in the compiled jar and so is specific to it.

We must also update the daffodil-cli/bin.LICENSE and daffodil-cli/bin.NOTICE files if the daffodil-schematron jar (and it's dependencies) will will be bundled in that CLI zip/tar/rpm/msi.

I think you also mentioned some of this came from Apache Camel? If so, we need to add the camel NOTICE information to the root NOTICE file, and the NOTICE file in ``daffodil-schemtron/src/main/resources/METAINF/NOTICE``. But it feels like maybe Apache Camel did not do the licesning correctly? We can't simply add an Apache license to something that is already MIT licensed.

Reply:
> Did you add the Apache license or did you get it from Apache Camel like that?

It does look like the Camel project adds the Apache header.  I had only compared the manifest from Camel with the ISO Schematron 2 distribution, not the file contents.  The best way from here is to replace all of these with the files from the official distribution, which only have the MIT header.

The Extract* files are a special case, not part of the distribution.  Will dig deeper for authoritative source on those, I see a couple different versions floating around.  Ill pull those out and put in the next PR, which makes more sense anyways.

> I think you also mentioned some of this came from Apache Camel?

The resources came from over there, but are going to be replaced with the standard distribution, nothing else remains a direct use.  Overall I used their approach to guide the Scala implementation so wanted to give credit in readme, but everything was converted from Java to Scala and heavily changed.

### todo
- [ ] remove the Extract resources
- [ ] replace other resources with ISO Sch distribution

I can address this later today

Reply:
The schematron readme and site does say it is available under MIT _and_ Apache licenses

https://schematron.com/front-page/the-schematron-skeleton-implementation/

and in the schematron repo there are various parts that are licensed under apache 2.0, but not these resources.

Im not sure what it all means, but still seems like the simplest thing to do is replace all the files using the distro zip.

Reply:
Yeah, I'd say it's safest to just get things directly from the source. Following your link, it looks like https://github.com/schxslt/schxslt has become the latest for the XSLT schematron implementation. That doesn't mention anything about Apache license, but it's MIT which is fine as long as we update the various LICENSE files appropriately. Let me know if anything is unclear about where to make license updates.

Though, looking some more at schxslt, it looks like there's also some java classes for it:

https://github.com/schxslt/schxslt-java

Which appears to be available on maven:

https://search.maven.org/artifact/name.dmaus.schxslt/schxslt/1.5.2/jar

That's also MIT licensed--it might be worth using that library to replace some the code in this commit. The less we have to maintain the better.  A quick test shows it doesn't even pull in any dependencies I think, which is nice.

Reply:
Yeah saw that java lib.

>  A quick test shows it doesn't even pull in any dependencies

Hmm

https://github.com/schxslt/schxslt-java/blob/master/pom.xml#L93-L109

https://github.com/schxslt/schxslt-java/blob/master/pom.xml#L123-L128

Reply:
Weird. When I tested it I added ``"name.dmaus.schxslt" % "schxslt" % "1.5.2"`` as a depenedency in our build.sbt and ran "sbt dependencyTree" and it didn't show any new transitive dependencies. Maybe these dependencies need to be manually added? Not sure. Whatever you think is fine with me.